patches/tModLoader/Terraria/ModLoader/ItemLoader.cs

```diff
@@ -1431,89 +1431,107 @@ public static bool CanStackInWorld(Item increase, Item decrease)
+	/// Attempts to stack <paramref name="destination"/> onto <paramref name="source"/>
 	/// </summary>
-	/// <param name="increase">Item where the stack is being increased.</param>
-	/// <param name="decrease">Item where the stack is being reduced.</param>
-	/// <param name="numTransfered">Amount to be transfered </param>
-	/// <param name="infiniteSource"></param>
-	/// <param name="numToTransfer">Used to only transfer a specidied amount instead of all.</param>
-	public static void StackItems(Item increase, Item decrease, out int numTransfered, bool infiniteSource = false, int? numToTransfer = null)
+	/// <param name="destination">The item instance that <paramref name="source"/> will attempt to stack onto</param>
+	/// <param name="source">The item instance being stacked onto <paramref name="destination"/></param>
+	/// <param name="numTransferred">The quanity of <paramref name="source"/> that was transferred to <paramref name="destination"/></param>
+	/// <param name="infiniteSource">Whether <paramref name="source"/>.stack should be decreased</param>
+	/// <param name="numToTransfer">
+	/// An optional argument used to specify the quantity of items to transfer form <paramref name="source"/> to <paramref name="destination"/>.<br/>
+	/// By default, the transfer quantity is the minimum between <paramref name="source"/>.stack and <paramref name="destination"/>.maxStack - <paramref name="destination"/>.stack
+	/// </param>
+	public static void StackItems(Item destination, Item source, out int numTransferred, bool infiniteSource = false, int? numToTransfer = null)
 	{
-		numTransfered = numToTransfer ?? Math.Min(decrease.stack, increase.maxStack - increase.stack);
+		numTransferred = numToTransfer ?? Math.Min(source.stack, destination.maxStack - destination.stack);
 
-		foreach (var g in HookOnStack.Enumerate(increase.globalItems)) {
-			g.OnStack(increase, decrease, numTransfered);
+		OnStack(destination, source, numTransferred);
+
+		if (source.favorited) {
+			destination.favorited = true;
+			source.favorited = false;
 		}
 
-		increase.ModItem?.OnStack(decrease, numTransfered);
+		destination.stack += numTransferred;
+		if (!infiniteSource)
+			source.stack -= numTransferred;
+	}
 
-		if (decrease.favorited) {
-			increase.favorited = true;
-			decrease.favorited = false;
+	/// <summary>
+	/// Calls the GlobalItem.OnStack hooks in <paramref name="destination"/>, then the ModItem.OnStack hook in <paramref name="destination"/>
+	/// </summary>
+	/// <param name="destination">The item instance that <paramref name="source"/> will attempt to stack onto</param>
+	/// <param name="source">The item instance being stacked onto <paramref name="destination"/></param>
+	/// <param name="numTransferred">The quanity of <paramref name="source"/> that was transferred to <paramref name="destination"/></param>
+	public static void OnStack(Item destination, Item source, int numTransferred)
```

Comment:
Documentation and naming of `numTransferred` param is inconsistent with actual hooks. `numToTransfer`. The transfer hasn't actually happened yet. Should make this clear in the summary.

Technically the hook is more like `OnAboutToStack`